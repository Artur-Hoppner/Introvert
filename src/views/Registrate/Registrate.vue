<template>
  <v-content>
    <v-container class="container">
      <v-layout>
        <v-flex class="login-form text-xs-center">
          <div class="display-1 mb-3 ">
            <v-icon x-large>assignment_ind</v-icon> Sign Up
          </div>
          <v-card>
            <v-card-text>
              <div>
                <template>Create your account</template>
              </div>
              <v-form @keyup.enter.native="registration">
                <v-text-field
                  v-model="username"
                  prepend-icon="person"
                  label="Username"
                ></v-text-field>
                <v-text-field
                  v-model="name"
                  prepend-icon="person"
                  label="Name"
                ></v-text-field>
                <v-text-field
                  v-model="email"
                  light="light"
                  prepend-icon="email"
                  label="Email"
                  type="text"
                ></v-text-field>
                <v-text-field
                  v-model="password"
                  light="light"
                  prepend-icon="lock"
                  label="Password"
                  type="password"
                ></v-text-field>
                <v-text-field
                  v-model="verifypassword"
                  light="light"
                  prepend-icon="lock"
                  label="Verify Password"
                  type="password"
                ></v-text-field>

                <v-btn @click="registration">Sign up</v-btn>
                <v-dialog v-model="dialog" width="600px">
                  <template v-slot:activator="{ on, attrs }">
                    <v-btn light v-bind="attrs" v-on="on">
                      Information
                    </v-btn>
                  </template>
                  <v-card>
                    <v-card-title>
                      <span class="headline">Use of stored data:</span>
                    </v-card-title>
                    <v-card-text
                      >We processing data in accordance with its
                      responsibilities under the GDPR.
                    </v-card-text>
                    <v-card-actions>
                      <v-spacer></v-spacer>
                    </v-card-actions>
                  </v-card>
                </v-dialog>
              </v-form>
            </v-card-text>
          </v-card>
        </v-flex>
      </v-layout>
    </v-container>
  </v-content>
</template>

<script>
import { mapActions } from 'vuex';
import { mapFields } from 'vuex-map-fields';
export default {
  name: 'Registration',
  computed: {
    ...mapFields([
      'newUser.name',
      'newUser.username',
      'newUser.email',
      'newUser.password',
      'newUser.verifypassword'
    ]),
    storvalue() {
      return this.$store.state.newUser;
    }
  },
  methods: {
    ...mapActions(['registration'])
  }
};
</script>

<style lang="scss" scoped>
.container {
  max-width: '20px';
}
.sign-up {
  margin: '10px';
}
</style>
