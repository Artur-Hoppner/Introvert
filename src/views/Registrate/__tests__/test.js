// Utilities
import { mount, createLocalVue } from '@vue/test-utils';
// Libraries
import Vue from 'vue';
import Vuex from 'vuex';
import VueRouter from 'vue-router';
import Vuetify from 'vuetify';
// Components
import Registrate from '@/views/Registrate/Registrate.vue';

const localVue = createLocalVue();
localVue.use(Vuex, VueRouter);

jest.mock('vuex-map-fields', () => ({
  mapFields: jest.fn()
}));

describe('/registrate', () => {
  let vuetify;
  beforeEach(() => {
    Vue.use(Vuetify);
    vuetify = new Vuetify();
  });
  test('Test id Registrate is rendered', () => {
    // Arrange;
    const wrapper = mount(Registrate, {
      localVue,
      vuetify
    });
    // Assert
    expect(wrapper.element).toMatchSnapshot();
  });

  test('It should activate registration on click', () => {
    const actions = {
      registration: jest.fn()
    };
    const store = new Vuex.Store({
      actions
    });

    // Arrange;
    const wrapper = mount(Registrate, {
      localVue,
      vuetify,
      store
    });

    const button = wrapper.find('.v-btn');
    expect(actions.registration).toHaveBeenCalledTimes(0);
    button.trigger('click');
    expect(actions.registration).toHaveBeenCalledTimes(1);
  });
});
