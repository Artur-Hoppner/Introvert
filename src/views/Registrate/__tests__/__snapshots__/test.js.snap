// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/registrate Test id Registrate is rendered 1`] = `
<div
  class="container container"
>
  <div
    class="layout"
  >
    <div
      class="flex login-form text-xs-center"
    >
      <div
        class="display-1 mb-3 "
      >
        <i
          aria-hidden="true"
          class="v-icon notranslate material-icons theme--light"
          style="font-size: 40px;"
        >
          assignment_ind
        </i>
         Sign Up
      
      </div>
       
      <div
        class="v-card v-sheet theme--light"
      >
        <div
          class="v-card__text"
        >
          <div>
            Create your account
          </div>
           
          <form
            class="v-form"
            novalidate="novalidate"
          >
            <div
              class="v-input theme--light v-text-field"
            >
              <div
                class="v-input__prepend-outer"
              >
                <div
                  class="v-input__icon v-input__icon--prepend"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate material-icons theme--light"
                  >
                    person
                  </i>
                </div>
              </div>
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label theme--light"
                      for="input-5"
                      style="left: 0px; position: absolute;"
                    >
                      Username
                    </label>
                    <input
                      id="input-5"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <transition-group-stub
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="v-input theme--light v-text-field"
            >
              <div
                class="v-input__prepend-outer"
              >
                <div
                  class="v-input__icon v-input__icon--prepend"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate material-icons theme--light"
                  >
                    person
                  </i>
                </div>
              </div>
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label theme--light"
                      for="input-9"
                      style="left: 0px; position: absolute;"
                    >
                      Name
                    </label>
                    <input
                      id="input-9"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <transition-group-stub
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="v-input theme--light v-text-field"
            >
              <div
                class="v-input__prepend-outer"
              >
                <div
                  class="v-input__icon v-input__icon--prepend"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate material-icons theme--light"
                  >
                    email
                  </i>
                </div>
              </div>
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label theme--light"
                      for="input-13"
                      style="left: 0px; position: absolute;"
                    >
                      Email
                    </label>
                    <input
                      id="input-13"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <transition-group-stub
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="v-input theme--light v-text-field"
            >
              <div
                class="v-input__prepend-outer"
              >
                <div
                  class="v-input__icon v-input__icon--prepend"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate material-icons theme--light"
                  >
                    lock
                  </i>
                </div>
              </div>
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label theme--light"
                      for="input-17"
                      style="left: 0px; position: absolute;"
                    >
                      Password
                    </label>
                    <input
                      id="input-17"
                      type="password"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <transition-group-stub
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="v-input theme--light v-text-field"
            >
              <div
                class="v-input__prepend-outer"
              >
                <div
                  class="v-input__icon v-input__icon--prepend"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate material-icons theme--light"
                  >
                    lock
                  </i>
                </div>
              </div>
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label theme--light"
                      for="input-21"
                      style="left: 0px; position: absolute;"
                    >
                      Verify Password
                    </label>
                    <input
                      id="input-21"
                      type="password"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <transition-group-stub
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
             
            <button
              class="v-btn v-btn--contained theme--light v-size--default"
              type="button"
            >
              <span
                class="v-btn__content"
              >
                Sign up
              </span>
            </button>
             
            <div
              class="v-dialog__container"
              role="dialog"
            >
              <button
                aria-expanded="false"
                aria-haspopup="true"
                class="v-btn v-btn--contained theme--light v-size--default"
                role="button"
                type="button"
              >
                <span
                  class="v-btn__content"
                >
                  
                  Information
                
                </span>
              </button>
              <!---->
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
